### ISSUE TITLE
Build 237 not consistently saving chunk data

### ISSUE THREAD

**Issue by MCPhssthpok** [PaperSpigot/Paper#8509]
### Expected behavior

When the server is restarted or when players leave an area long enough for an auto-save to occur, we expect the chunk data to be written back to disk.

### Observed/Actual behavior

Players have been reporting that changes they have made in the world have been backed out after a restart or when they've been away from the area or logged out. The unsaved changes are aligned to chunk boundaries.

This problem started yesterday after the server was updated to build 237 which contains changes relating to chunk saving. I know this is only circumstantial but I have not been able to find any problems with the relevant chunks (I ran the whole world save through the Region Fixer utility), the server has plenty of storage space and there is nothing reported in the server logs.

### Steps/models to reproduce

Not sure on how to reproduce as the problem is intermittent.

### Plugin and Datapack List

![paper-237-plugins](https://user-images.githubusercontent.com/92108740/197997031-243cbbd7-5dc0-45f2-a1bc-57a7b2874ccd.png)
![paper-237-datapacks](https://user-images.githubusercontent.com/92108740/197997067-ebeda855-61fe-445b-8624-1870dfa744b4.png)

### Paper version

![paper-237-version](https://user-images.githubusercontent.com/92108740/197997120-72d09272-995e-47e0-acb9-26706da6ef5e.png)

**Comment by joshuamartin11** [comment]
Is it possible that the chunks reset whilst the server was in build #235? Chunk saving was broken in build #233 but was meant to be fixed in #237.

**Comment by MCPhssthpok** [comment]
It is more than possible now that I stop panicking and look more closely. The first report I got was at 9am BST 25th when the server was still running build 235. I got more reports after a restart at 6pm BST when the server was updated to build 237 but of course that would still have been on 235 when it shut down and saved the world.

I'll keep an eye on things and close this issue if nothing more comes up.
